### PR TITLE
Add wprojection operator

### PR DESCRIPTION
**Pull request by Luke-Pratley**
Past work that needs to be added before release

### CODE REVIEW

File: cpp/example/padmm_random_coverage.cc [hunk 1]
```diff
@@ -129,24 +130,23 @@ int main(int, char **) {
...
-  const t_real snr = 10;
+  const t_real FoV = 15;       // deg
+  const t_real max_w = 15.;  // lambda
+  const t_real snr = 30;
   const bool w_term = false;
   const std::string kernel = "kb";
   std::string const fitsfile = image_filename(name + ".fits");
   auto M31 = pfitsio::read2d(fitsfile);
-  const t_real cellsize = 1;  // FoV / M31.cols() * 60. * 60.;
+  const t_real cellsize = FoV / M31.cols() * 60. * 60.;
```
Comment (dpshelio) [inline cpp/example/padmm_random_coverage.cc]: Is this correct? if it's so, proably I would find it more readable as: 
```suggestion
  const t_real cellsize = (FoV * 60. * 60.)/ 31.cols(); // [arcsecs/px]
```

Comment (Luke-Pratley) [reply cpp/example/padmm_random_coverage.cc]: Actually, I think this formula is only right for small fields of view (FoV). I wrote a function that will work though...

File: cmake_files/LookUpCubature.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Installs stevengj/Cubature into build directory
+#
+# - GIT_REPOSITORY: defaults to origin stevengj repo on github
+# - GIT_TAG: defaults to master
+if(Cubature_ARGUMENTS)
+  cmake_parse_arguments(Cubature "" "GIT_REPOSITORY;GIT_TAG" ""
+    ${Cubature_ARGUMENTS})
+endif()
+if(NOT Cubature_GIT_REPOSITORY)
+  set(Cubature_GIT_REPOSITORY https://github.com/stevengj/cubature)
```
[the remark this answers is not in the document]
Comment (Luke-Pratley) [reply cmake_files/LookUpCubature.cmake]: @dpshelio I wrote unit tests to test the library, at least the wrappers I wrote for it 😅😅

File: cpp/main.cc [hunk 3]
```diff
@@ -77,57 +113,129 @@ int main(int argc, const char **argv) {
...
+                  mop_algo, uv_data, params.height(), params.width(), params.cellsizey(),
+                  params.cellsizex(), params.oversampling(),
+                  kernels::kernel_from_string.at(params.kernel()), 2 * params.Jy(), 2 * params.Jx(),
+                  params.mpi_wstacking())
+            : factory::measurement_operator_factory<Vector<t_complex>>(
+                  mop_algo, uv_data, params.height(), params.width(), params.cellsizey(),
+                  params.cellsizex(), params.oversampling(),
+                  kernels::kernel_from_string.at(params.kernel()), 2 * params.Jx(), params.Jw(),
+                  1e-6, 1e-6);
+#ifndef PURIFY_MPI
```
Comment (dpshelio) [inline cpp/main.cc]: Shouldn't this be
```suggestion
#ifdef PURIFY_MPI
```

**Comment by dpshelio** [comment]
It passes!!! Can we merge???

**Comment by Luke-Pratley** [comment]
@dpshelio After this last commit!

**Pull request merged**